compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/scene/ComposeScene.skiko.kt

```diff
@@ -117,6 +118,11 @@ interface ComposeScene {
      */
     val focusManager: ComposeSceneFocusManager
 
+    /**
+     * Returns the currently active [PlatformDragAndDropManager].
+     */
+    fun activeDragAndDropManager(): PlatformDragAndDropManager
```

Comment:
What is the reasoning behind having a function instead of a property? Users of this API don't need to know that there is an implementation that contains multiple owners/managers

Reply:
```suggestion
    val dragAndDropManager: PlatformDragAndDropManager
```

Reply:
A property has the connotation of being simple value to calculate. A function is more generic with no such connotations, and so the user is more likely to avoid calling it twice in a row instead of reusing the already obtained value.

Additionally, a property typically returns the same value over invocations.

https://kotlinlang.org/docs/coding-conventions.html#functions-vs-properties

In this case we have implementations that iterate over an array, and return different values at different times. So a function is more appropriate.

Reply:
While I partially agree with it for the generic case, I don't think that it fits here. Even in general, the property doesn't have a semantic guarantee of a stable, non-calculated value.

In this particular case you want to "expose" some implementation details, but the semantic of it is clearly a property

Even from the link you posted, we have 3/3 cases why it should be a property

- [x] does not throw
- [x] is cheap to calculate (non-calculated in one implementation and for other in most of the cases we have just 1-2 layers)
- [x] returns the same result over invocations **if the object state hasn't changed** (different value ONLY if state changed)

Reply:
> is cheap to calculate (non-calculated in one implementation and for other in most of the cases we have just 1-2 layers)
1. That's an implementation detail.
2. The number of layers isn't really known.

- This is an interface, and this isn't an *obvious* property, so we should default to a function. 
- Using `val` can give the caller the wrong impression about this, possibly encouraging two mistakes:
1. Assuming it's a cheap call, calling it several times unnecessarily.
2. Assuming it can't change at any time, writing code like this:
   ```
   val dndManager = scene.dragAndDropManager
   // long running task here
   dndManager.doSomething()
   ```
Making it a function does not bring these mistaken connotations.

Some examples from our code:
- `PointerIconService.getIcon()`
- `Selectable.getText()`
- `ComposeScene.hasInvalidations()`.

Reply:
> That's an implementation detail

Exactly. This is why it should be a property

> Assuming it can't change at any time

It's not the case. The case is "if the object state hasn't changed"

It's the predictable value of the current state, I don't see a reason to make this as a function

Reply:
> Exactly. This is why it should be a property

It's an implementation detail whether it's cheap. It could be non-cheap. Hence it should not be a `val`.

> It's not the case. The case is "if the object state hasn't changed"

This is not an important distinction. Someone using `ComposeScene` via its interface isn't aware of all the possible state changes it could have. `val` brings with it connotations of a stable/unchanging property, which is wrong here and can encourage wrong assumptions.

Reply:
This interface is hiding implementation details. **Including** the fact that it can have multiple owners/layers inside. Exposing this and forcing users to think about it seems wrong to me.

Reply:
A function is the most generic thing, that doesn't carry any extra connotations. A property does carry some.

I'm not exposing here the fact that there are multiple owners/layers inside. I'm exposing the fact that there *could* be (something; not specifically owners/layers), and the caller should not assume there aren't.

Reply:
For me it still looks quite unnatural to have the function instead of a property here. Also, I believe that property fits all criteria from Kotlin conventions.
But all of it I already wrote before. Since we have no compatibility guarantees here, I won't block the change because of it. But I consider it as a subject of future adjustments.